### PR TITLE
update setroledef command 

### PR DESCRIPTION
**Pull request by darshanhs90**
…not on scopes

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@darshanhs90 looks like a couple of tests need to be re-recorded

File: src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs [hunk 1]
```diff
@@ -595,7 +595,16 @@ public PSRoleDefinition UpdateRoleDefinition(PSRoleDefinition roleDefinition)
 
             ValidateRoleDefinition(roleDefinition);
 
-            PSRoleDefinition fetchedRoleDefinition = this.GetRoleDefinition(roleDefinitionId, roleDefinition.AssignableScopes.First());
+			PSRoleDefinition fetchedRoleDefinition = null;
+			foreach (String scope in roleDefinition.AssignableScopes)
```
Comment (cormacpayne) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs]: @darshanhs90 is this change covered in existing tests?

Comment (darshanhs90) [inline src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs]: @cormacpayne added tests for this

**Comment by cormacpayne** [comment]
@darshanhs90 please take a look at the compilation errors

**Comment by darshanhs90** [comment]
@cormacpayne fixed the build issues

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleDefinitionTests.ps1 [hunk 2]
```diff
@@ -128,6 +128,49 @@ function Test-RDPositiveScenarios
 
 <#
 .SYNOPSIS
+Tests verify roledefinition update with interchanged assignablescopes.
+#>
+function Test-RDUpdate
+{
+    # Setup
+    Add-Type -Path ".\\Microsoft.Azure.Commands.Resources.dll"
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleDefinitionTests.ps1]: Why are you adding the cmlets this way?  They should be added through your test controller (and likely already are)

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleDefinitionTests.ps1 [hunk 3]
```diff
@@ -128,6 +128,49 @@ function Test-RDPositiveScenarios
...
+}
+
+<#
+.SYNOPSIS
+Tests verify roledefinition create with invalid scope.
+#>
+function Test-RDCreateFromFile
+{
+    # Setup
+    Add-Type -Path ".\\Microsoft.Azure.Commands.Resources.dll"
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleDefinitionTests.ps1]: Again, this should not be needed

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1805/

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleDefinitionTests.ps1 (hunk 2 above)
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/RoleDefinitionTests.ps1]: Filed issue: #4411 to address this for all Authorization tests

**Pull request merged**